pyop2/compilation.py

```diff
@@ -54,6 +55,31 @@ def _check_hashes(x, y, datatype):
     return False
 
 
+@collective
+def single_out_a_rank(comm):
+    rank = MPI.Comm.Get_rank(comm)
+    debug('Using file-based mechanism on %d', rank)
+    fname = os.path.join("/tmp", "fb.%d" % rank)
+
+    with open(fname, 'w+') as f:
+        f.write("This is a file.")
+
+    MPI.Comm.Barrier(comm)
+    tbb = []
+    # for dl, sl, fl in os.walk("/tmp"):
+    #     for fi in fl:
+    files = glob.glob("/tmp/fb.[0-9]+")
+    for fi in files:
+        if str(fi).split(".")[0] == "fb":
+            tbb.append(str(fi).split(".")[1])
+
+    os.remove(fname)
+    if str(min(tbb)) == str(rank):
```

Comment:
It seems to be that `tbb` is a list of strings rather than a list of integers, so this will lexicographically smallest number rather than the smallest number. Though this shouldn't cause a problem.